helix-term/src/ui/completion.rs

```diff
@@ -381,104 +381,106 @@ impl Component for Completion {
 
         // if we have a selection, render a markdown popup on top/below with info
-        if let Some(option) = self.popup.contents().selection() {
-            // need to render:
-            // option.detail
-            // ---
-            // option.documentation
-
-            let (view, doc) = current!(cx.editor);
-            let language = doc.language_name().unwrap_or("");
-            let text = doc.text().slice(..);
-            let cursor_pos = doc.selection(view.id).primary().cursor(text);
-            let coords = view
-                .screen_coords_at_pos(doc, text, cursor_pos)
-                .expect("cursor must be in view");
-            let cursor_pos = coords.row as u16;
-
-            let mut markdown_doc = match &option.documentation {
-                Some(lsp::Documentation::String(contents))
-                | Some(lsp::Documentation::MarkupContent(lsp::MarkupContent {
-                    kind: lsp::MarkupKind::PlainText,
-                    value: contents,
-                })) => {
-                    // TODO: convert to wrapped text
-                    Markdown::new(
-                        format!(
-                            "```{}\n{}\n```\n{}",
-                            language,
-                            option.detail.as_deref().unwrap_or_default(),
-                            contents
-                        ),
-                        cx.editor.syn_loader.clone(),
-                    )
-                }
-                Some(lsp::Documentation::MarkupContent(lsp::MarkupContent {
-                    kind: lsp::MarkupKind::Markdown,
-                    value: contents,
-                })) => {
-                    // TODO: set language based on doc scope
-                    if let Some(detail) = &option.detail.as_deref() {
-                        Markdown::new(
-                            format!("```{}\n{}\n```\n{}", language, detail, contents),
-                            cx.editor.syn_loader.clone(),
-                        )
-                    } else {
-                        Markdown::new(contents.to_string(), cx.editor.syn_loader.clone())
-                    }
-                }
-                None if option.detail.is_some() => {
-                    // TODO: copied from above
-
-                    // TODO: set language based on doc scope
+        let option = match self.popup.contents().selection() {
+            Some(option) => option,
+            None => return,
+        };
+        // need to render:
+        // option.detail
+        // ---
+        // option.documentation
+
+        let (view, doc) = current!(cx.editor);
+        let language = doc.language_name().unwrap_or("");
+        let text = doc.text().slice(..);
+        let cursor_pos = doc.selection(view.id).primary().cursor(text);
+        let coords = view
+            .screen_coords_at_pos(doc, text, cursor_pos)
+            .expect("cursor must be in view");
```

Comment:
will it panic?

Reply:
The primary cursor will always be on screen, so it shouldn't panic. Also the `expect()` call was already part of the code before the change (turning on ignore whitespace changes in the diff UI will give a more accurate view of the change set).

Reply:
it was introduced in #5420. It originally used the `visual_offset_at_pos` function which always returned a value as that function is not aware of the view position. With #5420 all those occurances had to be changed to use view position aware. I am pretty sure that `expect` is fine here because the cursor should always be in view while editing in insert mode. 

There are situations where the cursor is not inside the view (mostly bugs/oversights but it happens on purpose too) but I am pretty sure that's never the case while editing in insert mode (barring bugs). I am not even sure what else we could o here if the cursor is not in screen? Just not draw the popup?